### DISCUSSION TITLE
Docker image tag chaos for rtl_433

### DISCUSSION THREAD

**Discussion by LordOfTheSnow** [hertzg/rtl_433_docker#58] [Q&A] [answered]
Hi,

can anybody please advise me what the the literally dozens of different docker image tags mean?

For example what is the difference between hertzg/rtl_433:alpine3-21.05 and hertzg/rtl_433:alpine-3-21.05? And this is just one of probably 100 examples you can find.

The tag naming here is an absolute nightmare and I am totally lost what the latest alpine-3 version is.

**Comment by hertzg** [comment, marked as answer]
Hello, 

Sure, It might be overwhelming for some people to see so many tags, I'd be happy to review prs with improved docs regarding the tags.
I'll try to explain better here.

The full format for any version is: `hertzg/rtl_433:<flavour>-<flavour_version>-<rtl433_version>`
There are aliases for alpine, for example container which is based on `alpine 3.15.0` (currently targeted latest version) would end generating 3 more tags for the same container (same hash).

* `hertzg/rtl_433:alpine-3.15.0-<rtl433_version>`
* `hertzg/rtl_433:alpine-3.15-<rtl433_version>` 
* `hertzg/rtl_433:alpine-3-<rtl433_version>`

The first `-` after the flavour name is optional for alpine and can be ommited which means the same image gets tagged with 3 more extra tags:

* `hertzg/rtl_433:alpine3.15.0-<rtl433_version>`
* `hertzg/rtl_433:alpine3.15-<rtl433_version>` 
* `hertzg/rtl_433:alpine3-<rtl433_version>`

Code snippet responsible for tagging each **Alpine** based build: 
Here `${alpine}` is the Alpine version, `${version}` is the `rtl_433` version.
`semMinor` and `semMajor` shorten the version up to `minor` and `major` versions respectively.

https://github.com/hertzg/rtl_433_docker/blob/36a056dfe5b788f17f44cd8d4746e679d7428242/images/alpine/config.js#L26-L36


Hope this helps :)

**Reply by LordOfTheSnow** [reply]
Thank you for the fast reply.

This helps at least with the alpine versions. Still there are much more versions that clutter the view. For example what are 20.11-alpine3, 0.1-alpine3, 0.01-alpine3, alpine3-20.11, alpine3-0.1, master-alpine3, alpine-3-0.1, ... _endless list of other tags_.

I would really appreciate if you can clean up this mess and only leave tags according to the schema you mentioned above.

**Reply by hertzg** [reply]
I would appreciate if you would provide to this repository instead of calling other peoples work a mess without even looking into history. If you look closely and put a tiny bit of effort instead of just complaining you might even realize what I mean by this.
